src/soilbiogeochem/SoilBiogeochemDecompCascadeMIMICSMod.F90

```diff
@@ -0,0 +1,1274 @@
+         t_soisno       => temperature_inst%t_soisno_col               , & ! Input:  [real(r8) (:,:)   ]  soil temperature (Kelvin)  (-nlevsno+1:nlevgrnd)       
+         o2stress_sat   => ch4_inst%o2stress_sat_col                   , & ! Input:  [real(r8) (:,:)   ]  Ratio of oxygen available to that demanded by roots, aerobes, & methanotrophs (nlevsoi)
+         o2stress_unsat => ch4_inst%o2stress_unsat_col                 , & ! Input:  [real(r8) (:,:)   ]  Ratio of oxygen available to that demanded by roots, aerobes, & methanotrophs (nlevsoi)
+         finundated     => ch4_inst%finundated_col                     , & ! Input:  [real(r8) (:)     ]  fractional inundated area                                
+         decomp_cpools_vr => soilbiogeochem_carbonstate_inst%decomp_cpools_vr_col , &  ! Input: [real(r8) (:,:,:) ] (gC/m3)  vertically-resolved decomposing (litter, cwd, soil) C pools
+         pathfrac_decomp_cascade => soilbiogeochem_carbonflux_inst%pathfrac_decomp_cascade_col                                                                    , &  ! Output: [real(r8) (:,:,:) ]  what fraction of C leaving a given pool passes through a given transition (frac)
+         rf_decomp_cascade       => soilbiogeochem_carbonflux_inst%rf_decomp_cascade_col                                                                          , & ! Input:  [real(r8)          (:,:,:) ]  respired fraction in decomposition step (frac)
+         w_scalar       => soilbiogeochem_carbonflux_inst%w_scalar_col , & ! Output: [real(r8) (:,:)   ]  soil water scalar for decomp                           
+         o_scalar       => soilbiogeochem_carbonflux_inst%o_scalar_col , & ! Output: [real(r8) (:,:)   ]  fraction by which decomposition is limited by anoxia   
+         cn_col         => soilbiogeochem_carbonflux_inst%cn_col       , & ! Output: [real(r8) (:,:)   ]  C:N ratio
+         decomp_k       => soilbiogeochem_carbonflux_inst%decomp_k_col , & ! Output: [real(r8) (:,:,:) ]  rate for decomposition (1./sec)
+         spinup_factor  => decomp_cascade_con%spinup_factor              & ! Input:  [real(r8)          (:)     ]  factor for AD spinup associated with each pool           
+         )
+
+      mino2lim = CNParamsShareInst%mino2lim
+
+      days_per_year = get_days_per_year()
+
+!     ! Set "decomp_depth_efolding" parameter
+!     decomp_depth_efolding = CNParamsShareInst%decomp_depth_efolding
+
+      ! translate to per-second time constant
+      k_frag = 1._r8 / (secspday * days_per_year * CNParamsShareInst%tau_cwd)
+
+     ! calc ref rate
+      if ( spinup_state >= 1 ) then
+         do fc = 1,num_soilc
+            c = filter_soilc(fc)
+            !
+            if ( abs(spinup_factor(i_met_lit) - 1._r8) .gt. .000001_r8) then
+               spinup_geogterm_l1(c) = spinup_factor(i_met_lit) * get_spinup_latitude_term(grc%latdeg(col%gridcell(c)))
+            else
+               spinup_geogterm_l1(c) = 1._r8
```

Comment:
You could make 1._r8 a parameter "one" because it's used so much. That's not at all critical though since numbers important to make parameters are more complex ones that are likely to change. "one" isn't likely to change, and as such it might be more clear to use 1._r8 rather than "one". But, just throwing that out there for you to think about.

Reply:
I feel like the important consideration here is: If you change one of these, do others need to be changed as well? If so, you should extract a parameter for each group that should be kept the same. If these are all truly supposed to just be the value 1 and would never change from that, then I feel that using a hard-coded 1 is more clear and I don't see any downsides of it myself. (Maybe it's just me, but when I see use of a parameter like `zero` or `one` in code, I am never satisfied with assuming that it is actually exactly 0 or 1, but instead feel I need to check to confirm that it's really exactly 0 or 1, and not something close to 0, like 0.000001 – so this just slows me down in reading & understanding code.)

Reply:
I like how @billsacks expressed it. If it's truly just what you would see as 1 in a math equation leave it as it is. If it's something that could need to change as a group so become 1.00000001 or something then express it as a constant. It's true I've seen code where I feel I have to check if one or zero is actually just 1.0_r8 and 0.0_r8. So thanks for articulating that clearly @billsacks.

Reply:
That helps, thanks. I will keep them as "1" to indicate that they should be addressed independent of each other if at some point they must change.